### ISSUE TITLE
OGM migrations

### ISSUE THREAD

**Issue by dtomicevic** [memgraph/gqlalchemy#17]
@jbajic, I've seen the OGM is [on the roadmap](https://github.com/memgraph/gqlalchemy/projects/1#card-63220850) and I wanted to propose that we also design a migration system for it, similar to [Alembic](https://alembic.sqlalchemy.org/en/latest/). Given that property graphs have a dynamic schema, this could help make the projects more maintainable and help with data consistency issues across project versions.

**Comment by jbajic** [comment]
Yep, I agree that would be an excellent tool to go along with OGM, but I am unsure if I would like to have it inside GQLAlchemy, it feels more like a separate tool than an extension of GQLAlchemy. What do you think?

**Comment by dtomicevic** [comment]
Yes, I feel like that is the right way to go. Also, [Alembic](https://github.com/sqlalchemy/alembic) is a separate project from SQLAlchemy.

**Comment by g-despot** [comment]
This is committed as part of the GQLAlchemy 1.2 release so I am closing the issue.

**Comment by gitbuda** [comment]
As far as I can see we haven't actually implemented migrations as a part of 1.1/1.2. Moving this back to open 😄

**Comment by alex-linx** [comment]
Hi, any status updates on this feature?
We are building a a project with memgraph using the gqlalchemy OGM, and we are assuming that we might have to the schema quite a bit. Would be great to have some tooling for that :)

**Comment by katarinasupe** [comment]
Hi @alex-linx, currently, we are not working on this feature, but it is definitely a plan to do it in the future. I can keep you updated on the release where this feature will be developed. It's nice to hear you are working on a project using OGM and thank you for reporting this, it will help us prioritize better. Can you tell me more about the project and how critical this feature is for your project?

**Comment by alex-linx** [comment]
@katarinasupe Hi, we are working on a project where we build a graph mapping out entities in a digital environment using a graph database. We define the types of nodes and relationships using the OGM.
We are still designing our system, and expect to be adding and changing the graph structure and schema of our nodes. Having an easy to use tool for doing schema migrations will definitely help, since otherwise we would need to

**Comment by katarinasupe** [comment]
@alex-linx I can update you on when we decide to prioritize this for development. Until then, can you [join our Discord server](https://www.discord.gg/memgraph) and send me DM so that I know it's you? It will be easier to communicate there, and you can ask anything else you need.
